bindings/ruby/src/operator.rs

```diff
@@ -141,6 +142,30 @@ impl Operator {
         let operator = rb_self.0.clone();
         Ok(Io::new(&ruby, operator, path, mode)?)
     }
+
+    /// Lists the directory.
+    pub fn list(ruby: &Ruby, rb_self: &Self, path: String) -> Result<Lister, Error> {
+        let lister = rb_self
+            .0
+            .clone()
+            .lister(&path)
+            .map_err(|err| Error::new(ruby.exception_runtime_error(), err.to_string()))?;
+
+        Ok(Lister::new(lister))
+    }
+
+    /// Lists the directory recursively.
+    pub fn scan(ruby: &Ruby, rb_self: &Self, path: String) -> Result<Lister, Error> {
```

Comment:
Hi, `scan` has been deprecated, and bindings should not expose such an API. Instead, the `list` function should accept an argument that allows setting `recursive: true`.